src/components/Modal/Modal.tsx

```diff
@@ -182,6 +178,11 @@ export function Modal({
     const {isMounted, status} = useTransitionStatus(context, {duration: TRANSITION_DURATION});
     const previousStatus = usePrevious(status);
 
+    const animateHeightProps = useAnimateHeight({
+        ref: !disableHeightTransition && refs.floating.current ? refs.floating : undefined,
+        enabled: open,
+    });
```

Comment:
With proposed changes this will be
```suggestion
    useAnimateHeight({
        ref: refs.floating,
        enabled: status === 'open' && !disableHeightTransition,
    });
```